app/store/slices/credentialFoyer.ts

```diff
@@ -28,18 +44,32 @@ const initialState: CredentialFoyerState = {
   pendingCredentials: [],
 };
 
+const stageCredentials = createAsyncThunk('credentialFoyer/stageCredentials', async (credentials: Credential[]) => {
+  const existingCredentials = await CredentialRecord.getAllCredentials();
+  const existingCredentialIds = existingCredentials.map(({ credential }) => credential.id);
+  
+  const pendingCredentials = credentials.map((credential) => {
+    if (existingCredentialIds.includes(credential.id)) {
```

Comment:
I'm probably missing something obvious, but shouldn't this be "if the incoming credential is _not_ included in existingCredentialIds"?

Reply:
So the logic here states "if the incoming credential _is_ included in existingCredentialIds" return a `PendingCredential` but with the initial `Accepted` status and `Duplicate` message. So if an existing credential is scanned, instead of erroring out, the status shows that it has previously been accepted (i.e. it's a duplicate). 

Otherwise, we return a `PendingCredential` with default status/message.

However, we can update this behavior if needed.

Reply:
@sethduffin - got it, thanks for the explanation!